### ISSUE TITLE
Themes

### ISSUE THREAD

**Issue by QarsoonMeel** [QarsoonMeel/DreamersDreams#5]
- add themes for each dream/quality like Lovecraft, myths and religion

**Comment by statecouncil** [comment]
Would you like me to add that in the dream inventory already?

**Comment by QarsoonMeel** [comment]
What I meant by themes is that I want to have control over what dreams appear in the game. For example someone doesn't like Lovecraft? Well you can turn off all dreams that relate to that. It is not dream quality but a dream theme.

Possible themes: Lovecraft, archotech, violent, drugs related etc.

**Comment by statecouncil** [comment]
Yeah, I get that. I actually already started doing that by labeling short dreams as "generic", but we can definitely expand that. I just fear that this feature will make managing the dreams quite a bit more difficult, once we reach larger amounts of dreams. On the other hand, this makes mod compatibility and adding further dreams easier, as we can kind of treat them like Cards Against Humanity expansions, or "packs". We discuss it next week?

**Comment by QarsoonMeel** [comment]
It will be important once we get people to help make their own dream packs :) How do you label them? I hope you are not editing any defNames! :D

**Comment by statecouncil** [comment]
So far I've only labelled them in the Dream Inventory. It makes sense to nail down some dedicated tags, though. Let's just not go too overboard with it. I think it makes sense to put some information into defNames, like dream type (generic, normal, recurring, chaindream, sleepwalking), which iteration it is (tribal/outlander version of the same dream) or if it's a super special extra important custom dream. Examples:
- DreamPottedPlantTribal
- DreamPottedPlantOutlander
- CDreamPottedPlant05
- RDreamPottedPlant
How does that sound?

**Comment by QarsoonMeel** [comment]
No, no we can't change defNames at all. If we change those it will throw lots of errors for users. Also putting it in there would be plain hard to deal with in general. Best and easiest solution would be to make <themes> tag and allow players to turn off/on those.

**Comment by statecouncil** [comment]
Well, I'm not saying we should change defnames all the time, but I think it is a good idea to go with a consistent naming convention, moving forward. Right now, the DreamDefs are named very inconsistently and don't really allow room for growth if we have 1000 dreams instead of less than a hundred. I do agree that we should have tags as well, though.

**Comment by QarsoonMeel** [comment]
defNames should not concern anyone really. These do not impact growth or anything but we could use a convention to make sure they are consistent (just for looks). These should not be used for any checks or anything really other than references in the code. Not a single part of Rimworld code ever checks if a gun is a gun based on the name of the def. Let's stick to convention Dream[NameOfTheDream]. If you already made changes then that's fine I will replace those names with old ones or keep the names that fix up names to follow the right convention. If you need tools to sort the dreams please, let's discuss that.

**Comment by statecouncil** [comment]
Fair enough. I've enhanced the Dream Inventory a bit so we can earmark changes there to be added into the code. So far, the only drastic change from Dream[NameOfDream] is GenericDream[NameOfDream] and SpecialDream[NameOfDream], which is used for short, generic dreams and sleepwalking dreams, respectively. Would you like me to change that back? I can do that before I send a pull request.
I don't think we need a dream browser or anything like that at this point, but I'll keep a close eye on everything staying ordered. This can quickly become a mess if we just keep adding dreams without any idea what's already in there, especially in regard of duplicates. I've already thrown out a few that I missed the first time around.

**Comment by QarsoonMeel** [comment]
I can change that it's fine.
